rfcs/0017-state-snapshotting.md

```diff
@@ -0,0 +1,244 @@
+## Summary
+[summary]: #summary
+
+This RFC provides information on taking state snapshots. These snapshots would
+give us a mechanism for more quickly reproducing edge cases and errors as they
+arise. Snapshots can be broken into three projects: Partial and full-offline
+snapshots (covered here in detail), and full-online snapshots which is
+out-of-scope for now. Partial snapshotting can be done first and in parallel
+with implementation of transition frontier persistance to make the full-offline
+version much quicker to implement afterwards.
+
+## Motivation
+
+[motivation]: #motivation
+
+As stated above, snaphotting parts of our application state will give us a
+mechanism for more quickly reproducing edge cases and errors as they arive.
+
+There are several stages. Each of which provides a checkpoint that gives us
+useful features.
+
+[partial]: #partial
+### Partial Snapshotting:
+
+Snapshot enough state to be able to produce the immediate next transition, and
```

Comment:
Producing the immediate next transition means proposing, right? Can you mention that in this paragraph?